docs/guides/mongodb/rotate-auth/rotateauth.md

```diff
@@ -0,0 +1,493 @@
+
+```
+**Verify Auth is rotated**
+```shell
+$ kubectl get mg -n demo mgo-quickstart -ojson | jq .spec.authSecret.name
+"mgo-quickstart-auth"
+$ kubectl get secret -n demo mgo-quickstart-auth -o=jsonpath='{.data.username}' | base64 -d
+root⏎                                                               
+$ kubectl get secret -n demo mgo-quickstart-auth -o=jsonpath='{.data.password}' | base64 -d
+09wZM.)t8kpwKF5z⏎                      
+```
+Also, there will be two more new keys in the secret that stores the previous credentials. The keys are `username.prev` and `password.prev`. You can find the secret and its data by running the following command:
+
+```shell
+$ kubectl get secret -n demo mgo-quickstart-auth -o go-template='{{ index .data "username.prev" }}' | base64 -d
+root⏎                                                                                                          
+$ kubectl get secret -n demo mgo-quickstart-auth -o go-template='{{ index .data "password.prev" }}' | base64 -d
+eR*W_mz6bjyZxeiG⏎                        
+```
+The above output shows that the password has been changed successfully. The previous username & password is stored for rollback purpose.
+#### 2. Using user created credentials
+
+At first, we need to create a secret with kubernetes.io/basic-auth type using custom username and password. Below is the command to create a secret with kubernetes.io/basic-auth type,
+
+```shell
+$  kubectl create secret generic quick-mg-user-auth -n demo \
+                                                                                          --type=kubernetes.io/basic-auth \
+                                                                                          --from-literal=username=mg-admin \
+                                                                                          --from-literal=password=mongodb-secret
+secret/quick-mg-user-auth created
+```
+Now create a `MongoDBOpsRequest` with `RotateAuth` type. Below is the YAML of the `MongoDBOpsRequest` that we are going to create,
+
+```shell
+apiVersion: ops.kubedb.com/v1alpha1
+kind: MongoDBOpsRequest
+metadata:
+  name: mgops-rotate-auth-user
+  namespace: demo
+spec:
+  type: RotateAuth
+  databaseRef:
+    name: mgo-quickstart
+  authentication:
+    secretRef:
+      name: quick-mg-user-auth
+  timeout: 5m
+  apply: IfReady
+```
+Here,
+
+- `spec.databaseRef.name` specifies that we are performing rotate authentication operation on `mgo-quickstart`cluster.
+- `spec.type` specifies that we are performing `RotateAuth` on postgres.
+- `spec.authentication.secretRef.name` specifies that we are using `quick-mg-user-auth` as `spec.authSecret.name` for authentication.
+
+Let's create the `MongoDBOpsRequest` CR we have shown above,
+
+```shell
+$ kubectl apply -f https://github.com/kubedb/docs/raw/v2025.6.30/docs/examples/mongodb/rotate-auth/rotate-auth-user.yaml
```

Comment:
By templeting, I meant somehting like this : 
`v2025.6.30` ->  `{{ .version }}`